calico-vpp-agent/routing/connectivity/vxlan.go

```diff
@@ -53,7 +53,21 @@ func (p *VXLanProvider) configureVXLANNodes() error {
 }
 
 func (p *VXLanProvider) RescanState() {
-	// TODO
+	p.log.Infof("Rescanning existing VXLAN tunnels")
+	p.vxlanIfs = make(map[string]uint32)
+	tunnels, err := p.vpp.ListVXLanTunnels()
+	if err != nil {
+		p.log.Errorf("Error listing VXLan tunnels: %v", err)
+	}
+	nodeIP4 := p.server.GetNodeIP(false)
+	nodeIP6 := p.server.GetNodeIP(true)
+	for _, tunnel := range tunnels {
+		if tunnel.SrcAddress.Equal(nodeIP4) || tunnel.SrcAddress.Equal(nodeIP6) {
```

Comment:
Looking at this again, I think we need an additional check here on the VNI, and the ports as well. Other controllers (such as NSM) may create VXLAN tunnels in VPP with the same source IP but a different port / VNI, and we must not use these.